mavlink/modules/flight_controller.py

```diff
@@ -276,3 +277,27 @@ def download_commands(self) -> "tuple[bool, list[dronekit.Command]]":
         except ConnectionResetError:
             print("ERROR: Connection with drone reset. Unable to download commands.")
             return False, []
+
+    def get_next_waypoint(self) -> "tuple[bool, dronekit.Command | None]":
+        """
+        Gets the next waypoint.
+
+        Returns
+        -------
+        tuple[bool, dronekit.Command | None]
+        A tuple where the first element is a boolean indicating success or failure,
+        and the second element is the command currently held by the drone.
+        """
+        result, commands = self.download_commands()
+        if not result:
+            return False, None
+
+        next_command_index = self.drone.commands.next
+        if next_command_index >= len(commands):
+            return False, None
+
+        for i in range(next_command_index, len(commands)):
```

Comment:
maybe track the index in the class instead of using .next since it's easier to check index < length